src/mdbExtensionController.ts

```diff
@@ -520,24 +527,25 @@ export default class MDBExtensionController implements vscode.Disposable {
     // user yet, and they have no saved connections.
     if (!hasBeenShownViewAlready) {
       if (!this._storageController.hasSavedConnections()) {
-        vscode.commands.executeCommand(
+        void vscode.commands.executeCommand(
           EXTENSION_COMMANDS.MDB_OPEN_OVERVIEW_PAGE
         );
       }
 
-      this._storageController.update(
+      void this._storageController.update(
         StorageVariables.GLOBAL_HAS_BEEN_SHOWN_INITIAL_VIEW,
         true
       );
     }
   }
 
-  dispose(): void {
-    this.deactivate();
+  async dispose(): Promise<void> {
+    await this.deactivate();
   }
 
-  deactivate(): void {
-    this._connectionController.disconnect();
+  async deactivate(): Promise<void> {
+    await this._connectionController.disconnect();
```

Comment:
@Anemy do we actually need to wait for disconnecting before disposing of resources?

Reply:
What is the issue with waiting?